compiler/noirc_evaluator/src/ssa/opt/mem2reg.rs

```diff
@@ -444,7 +460,15 @@ impl<'f> PerFunctionContext<'f> {
     fn handle_terminator(&mut self, block: BasicBlockId, references: &mut Block) {
         self.inserter.map_terminator_in_place(block);
 
-        match self.inserter.function.dfg[block].unwrap_terminator() {
+        let terminator = self.inserter.function.dfg[block].unwrap_terminator();
+
+        terminator.for_each_value(|value| {
+            if let Some(context) = self.load_results.get_mut(&value) {
+                context.result_counter += 1;
+            }
```

Comment:
Makes me wonder if these checks all need to handle aliases as well. If we have multiple aliases this looks like it'd only increase the result counter for one of them.

Reply:
For now, I just check whether a reference has been aliased, and other I do not remove that store in the per function cleanup. In the case of just removing load results, it should be safe to remove a load if its result is not used anywhere even if an alias is being loaded. This counter is only being used to remove unused loads.